### ISSUE TITLE
Propose merging Projects view and Files view

### ISSUE THREAD

**Issue by AndyLuke** [apache/netbeans#6048]
### Description

Until now I can not understand why there are two views of the **Projects** view and **Files** view instead of a **Project Files** view. For example, I have a maven project root directory institutions include: src directory and help directory, pom.xml, LISENCE, README.md, package.xml (maven-assembly-plugin's package description file) and so on. As shown in the picture:
![Projects View](https://img1.imgtp.com/2023/06/06/02c1gHgh.png "Projects View")
![Files View](https://img1.imgtp.com/2023/06/06/valPjMH2.png "Files View")
Usually in coding development Projects view is the most common view, but this view only shows src directory and pom.xml file, I have to switch to Files view to see other files, but when in Files view, I have to open the class files step by step when I see them. To sum up, it is convenient to view class files in Projects view but not other files, and convenient to view other files in Files view but not class files. Therefore, it is highly recommended to merge Projects view and Files view to integrate the advantages of both, which is also the choice of most IDEs or code editors.

### Are you willing to submit a pull request?

No

**Comment by mbien** [comment]
you don't have to find the class again in the files view if you have it already open in the editor.

`ctrl+shift+1, 2 or 3` will select the file in projects, files or favorites view. The separation is intended. You might want to look at two places at once. Files is a pure filesystem view, Projects is not, it can show things which don't exist as files, like dependencies, or show files which are not in your project, like `.m2/settings.xml`

If you don't like using hot keys, you can also sync it automatically via `view -> synchronize editors with views` which is off by default.

**Comment by AndyLuke** [comment]
Thanks, although your suggestion can solve many of my problems, but what I want to express is that I want to improve the existing view presentation to make the IDE simpler and more powerful, and at the same time more friendly and convenient for developers. In short, simple and powerful, friendly and convenient.

**Comment by neilcsmith-net** [comment]
The distinction between [logical view](https://bits.netbeans.org/dev/javadoc/org-netbeans-modules-projectuiapi/org/netbeans/spi/project/ui/LogicalViewProvider.html) and the plain file view is a key facet of NetBeans' project support, and part of what makes it powerful.  I don't think we'd see a merging of these two concepts accepted.  I do think that some of the logical views could be enhanced somewhat so that use of the file tab is less necessary (eg. with Maven, showing readme, etc; accessing resource, assembly descriptors, and other locations even when the locations _don't_ exist on disk).  I would suggest considering what you think is missing in the project tab view, and take it from there.

> In addition, in Netbeans18, many shortcut keys in the shortcut key search are not available by name

Cannot reproduce, but you're the second person who has suggested this happening - please open another issue.

**Comment by AndyLuke** [comment]
> The distinction between [logical view](https://bits.netbeans.org/dev/javadoc/org-netbeans-modules-projectuiapi/org/netbeans/spi/project/ui/LogicalViewProvider.html) and the plain file view is a key facet of NetBeans' project support, and part of what makes it powerful.  I don't think we'd see a merging of these two concepts accepted.  I do think that some of the logical views could be enhanced somewhat so that use of the file tab is less necessary (eg. with Maven, showing readme, etc; accessing resource, assembly descriptors, and other locations even when the locations _don't_ exist on disk).  I would suggest considering what you think is missing in the project tab view, and take it from there.
> 
> > In addition, in Netbeans18, many shortcut keys in the shortcut key search are not available by name
> 
> Cannot reproduce, but you're the second person who has suggested this happening - please open another issue.

Yes, essentially it doesn't matter whether the two views are merged or not, the fundamental thing is that we want Netbeans to be more powerful, more convenient and better in terms of file display and manipulation.

**Comment by mbien** [comment]
> that we want Netbeans to be more powerful, more convenient and better in terms of file display and manipulation.

what does this mean?

**Comment by ebresie** [comment]
So my understanding of the two views are:

Project View - is project type specific with enhancements specific to a given project, allowing customized behavior based on the project type (i.e., Maven project allows maven dependency functionality, Java project allow "package based views", etc.) which may hide or arrange things in such a way as to simplify things for a given project type

File View - provides a  basic show the raw files within a given folder/project without any customizations, so it requires drilling down to specific folders to find a specific file instead of a "package" view, it shows all files (i.e., show a pom.xml where it's stored, .gitignore, etc.)

Favorites View - (not included in the original context but also helpful) identifies specific folders of interests; so specific project, sub-projects or sub-folders can be added and usable to navigate towards

So I think there are different use cases for each but once the usages is more familiar, I think it's good the way it is.

**Comment by AndyLuke** [comment]
> So my understanding of the two views are:
> 
> Project View - is project type specific with enhancements specific to a given project, allowing customized behavior based on the project type (i.e., Maven project allows maven dependency functionality, Java project allow "package based views", etc.) which may hide or arrange things in such a way as to simplify things for a given project type
> 
> File View - provides a  basic show the raw files within a given folder/project without any customizations, so it requires drilling down to specific folders to find a specific file instead of a "package" view, it shows all files (i.e., show a pom.xml where it's stored, .gitignore, etc.)
> 
> Favorites View - (not included in the original context but also helpful) identifies specific folders of interests; so specific project, sub-projects or sub-folders can be added and usable to navigate towards
> 
> So I think there are different use cases for each but once the usages is more familiar, I think it's good the way it is.

翻译结果

Yes, what you said makes sense, but I think the files in the project directory belong to the project itself, so why can't I see some of the project's files in the project view and have to switch views to find those files. In fact, my core point is to merge the functionality of the two views to complement each other, and the development team can make a comprehensive consideration as to whether to actually merge the two views. It is important to emphasize that the key word for the purpose of my proposal is **"convenience"**, because as of now, Netbeans is not really convenient for viewing files. This can be referred to Eclipse, Idea, VSCode and other IDEs.

**Comment by mbien** [comment]
>  files in the project directory belong to the project itself, so why can't I see some of the project's files in the project view

because the Projects view is no file browser it is a source code browser and contains nodes which don't even exist as files as previously mentioned. This is no view of the file system.

**Comment by AndyLuke** [comment]
> >  files in the project directory belong to the project itself, so why can't I see some of the project's files in the project view
> 
> because the Projects view is no file browser it is a source code browser and contains nodes which don't even exist as files as previously mentioned. This is no view of the file system.

Thank you, your explanation is very clear and I understand very well what you are saying. "files in the project directory belong to the project itself" is my personal opinion, because I think the project view should be able to view and manage all the files under the project.

You can disagree with me, but hopefully the file display can be improved to make it easier to use like IDEs like Eclipse.

**Comment by mbien** [comment]
> I think the project view should be able to view and manage all the files under the project.

unlikely that this is going to happen. All views have their purpose and can be used at the same time or switched between based on what you are doing.

**Comment by AndyLuke** [comment]
> > I think the project view should be able to view and manage all the files under the project.
> 
> unlikely that this is going to happen. All views have their purpose and can be used at the same time or switched between based on what you are doing.
> 
> edit: I could imagine for example making the "Important Files" node more configurable so that it can be customized on a per-project basis.

I think this solution is feasible to support custom display of files or directories under the per-project.

**Comment by Saljack** [comment]
I use this plugin https://github.com/Chris2011/ShowIgnoredFiles (it has some bugs but it is better then nothing) and I think a similar thing should be a part of the Projects view out of the box. There are many files which are connected to a project and I have to switch to Files. For example Spotbugs, PMD, `.gitignore`, `README.md`, pipeline configurations (`.gitlab-ci.yaml`), Kubernetes/helm configurations. 
It would be really nice to have an option to include all these files and folders to the Project Files or create a new node Other Files.

![image](https://github.com/apache/netbeans/assets/614424/5fbe34cb-0a77-4b05-9ad3-2a609267f12d)

**Comment by wumpz** [comment]
I strongly vote against a merge of project and files view. However, some improvements could be made:

* make additional files visible in project view (e.g. Jenkinsfile, ...) , there was a plugin doing that, but I don't find it anymore
* allow to filter projects view (e.g. hide dependency / test-dependency folder)
* somehow improve navigating project view with many open folder nodes (dependency, src), one is permanently scrolling
* add an actual project view: show only project for active file, or something like a subgroup for the actual project group, since project group switching is kind of slow
* dockable version of open documents dialog

Java 
* add more information into class icons of project view (visibility, final, module information)

**Comment by neilcsmith-net** [comment]
Moving this over to discussions.  There isn't anything specifically actionable here at the moment.  Various things discussed could be separate issues or PRs.
